src/Commands/MediaMpxCommands.php

```diff
@@ -0,0 +1,211 @@
+ *   - http://cgit.drupalcode.org/devel/tree/src/Commands/DevelCommands.php
+ *   - http://cgit.drupalcode.org/devel/tree/drush.services.yml
+ */
+class MediaMpxCommands extends DrushCommands {
+
+  /**
+   * The entity type manager used to load entities.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  private $entityTypeManager;
+
+  /**
+   * The factory used to load media data from mpx.
+   *
+   * @var \Drupal\media_mpx\DataObjectFactoryCreator
+   */
+  private $dataObjectFactoryCreator;
+
+  /**
+   * The factory used to store full mpx Media objects in the key-value store.
+   *
+   * @var \Drupal\Core\KeyValueStore\KeyValueFactoryInterface
+   */
+  private $keyValueFactory;
+
+  /**
+   * MediaMpxCommands constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The manager used to load config and media entities.
+   * @param \Drupal\media_mpx\DataObjectFactoryCreator $dataObjectFactoryCreator
+   *   The creator used to configure a factory for loading mpx objects.
+   * @param \Drupal\Core\KeyValueStore\KeyValueFactoryInterface $keyValueFactory
+   *   The key-value factory for storing complete objects.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, DataObjectFactoryCreator $dataObjectFactoryCreator, KeyValueFactoryInterface $keyValueFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->dataObjectFactoryCreator = $dataObjectFactoryCreator;
+    $this->keyValueFactory = $keyValueFactory;
+  }
+
+  /**
+   * Import mpx objects for a media type.
+   *
+   * @param string $media_type_id
+   *   The media type ID to import for.
+   *
+   * @usage media_mpx-import mpx_video
+   *   Import all mpx data for the mpx_video media type.
+   *
+   * @command media_mpx:import
+   * @aliases mpxi
+   */
+  public function import(string $media_type_id) {
+    $media_type = $this->loadMediaType($media_type_id);
+
+    $results = $this->selectAll($media_type);
+
+    // Store an array of media items we touched, so we can clear out their
+    // static cache.
+    $reset_ids = [];
+
+    // @todo Support fetching the total results via ObjectList.
+    $this->io()->title(dt('Importing @type media', ['@type' => $media_type_id]));
+    $this->io()->progressStart();
+
+    /** @var \Lullabot\Mpx\DataService\Media\Media $mpx_media */
+    foreach ($results as $index => $mpx_media) {
+      // @todo start a transaction.
+
+      // Find any existing media items, or return a new one.
+      $results = $this->loadMediaEntities($media_type, $mpx_media);
+
+      // Save the mpx Media item so it's available in getMetadata() in the
+      // source plugin.
+      $this->keyValueFactory->get('media_mpx_media')->set($mpx_media->getId(), $mpx_media);
+
+      foreach ($results as $media) {
+        $media->save();
+        $reset_ids[] = $media->id();
+      }
+
+      $this->entityTypeManager->getStorage('media')->resetCache($reset_ids);
```

Comment:
Why do you need to clear cache at this time? Are you updating results in the same loop?